storybook/README.md

```diff
@@ -1,17 +1,56 @@
-We are following the Component Story Format 3.
-Find more information about it here. https://storybook.js.org/blog/storybook-csf3-is-here
+# Goal
+Storybook is a tool for developing UI components in isolation. It makes building UIs organized and efficient.
+It also documents the components and their usage in an interactive way.
 
-If you are looking for a template from which to write stories, please do align with the ComposedChart.stories.tsx
+We want to migrate our current documentation to Storybook.
+The documentation is currently in the form of a static website, with embedded code sandboxes.
 
-- We want to use templates for rendering, and multiple stories with different parameters, content, data.
+## Structure
+As we want to migrate existing documentation to Storybook, we will want to reproduce the existing structure:
 
-- We want to document every single component in
--- the most simple manner
--- a complex usage
--- edge cases
+### API
+The API section contains stories for all components. The goal is to communicate a components behaviour as concise and simple as possible.
 
-- Additionally we will want to add stories that document complex usage of recharts with custom components.
+`storybook/stories/API/cartesian/Line.stories.tsx` can be considered a template for writing stories for the API section.
 
-Best practice: 
-- A story for a LineChart should not contain a complex brush. Each story should focus on a single component and its behaviour. If the interaction with the brush is to be shown, then any other brush details should ideally be omitted. 
-- Complex interaction effect between multiple component best deserve a separate story. 
+Acceptance criteria for API stories:
+- Each component should be documented in a doc file / mdx file.
```

Comment:
nit: can we just stick to not using `.mdx` to be consistent?

Reply:
It so turns out that for proper documentation we do need mdx. For an example I started here https://github.com/recharts/recharts/pull/3404